staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -233,40 +215,34 @@ var internalPackages = []string{"client-go/tools/cache/"}
 // RunWithStopOptions repeatedly uses the reflector's ListAndWatch to fetch all the
 // objects and subsequent deltas.
 // RunWithStopOptions will exit when one of the stopOptions conditions is met.
-func (r *Reflector) RunWithStopOptions(stopOptions StopOptions) {
-	// TODO: should we populate stop handle if nil here to avoid panic?
+func (r *Reflector) RunWithStopOptions(ctx context.Context, stopOptions StopOptions) {
 	klog.V(2).Infof("Starting reflector %s (%s) from %s", r.expectedTypeName, r.resyncPeriod, r.name)
-	if stopOptions.ExternalStop != nil {
-		cancel := r.stopHandle.MergeChan(stopOptions.ExternalStop)
-		defer cancel()
-		stopOptions.ExternalStop = nil
-	}
+	LWCtx, LWCancel := context.WithCancel(ctx)
 	wait.BackoffUntil(func() {
-		if err := r.ListAndWatch(r.stopHandle.Done()); err != nil {
+		if err := r.ListAndWatch(LWCtx.Done()); err != nil {
 			r.watchErrorHandler(r, err)
 			onListErr := stopOptions.OnListError
 			// default to never stopping (same as previous behavior that only had stop channel)
 			if onListErr != nil {
 				if stopOptions.OnListError(err) {
-					r.stopHandle.WithError(err)
-					klog.V(2).Infof("Closing with list error %v", r.stopHandle.Err())
-					r.stopHandle.Close()
+					klog.V(2).Infof("Closing with list err, type %s", r.expectedTypeName)
+					LWCancel()
 				}
 			}
 		}
-	}, r.backoffManager, true, r.stopHandle.Done())
+	}, r.backoffManager, true, LWCtx.Done())
 	klog.V(2).Infof("Stopping reflector %s (%s) from %s", r.expectedTypeName, r.resyncPeriod, r.name)
 }
 
 // Run calls RunWithStopOptions and only exits when stopCh is closed
```

Comment:
the comment say "Run calls RunWithStopOptions", but the code does not, why not?

Reply:
fixed, now it does.